lib/rubocop/cop/mixin/check_assignment.rb

```diff
@@ -17,27 +17,19 @@ def on_lvasgn(node)
       alias on_and_asgn on_lvasgn
 
       def on_send(node)
-        rhs = extract_rhs(node)
-
-        return unless rhs
+        return unless (rhs = extract_rhs(node))
 
         check_assignment(node, rhs)
       end
 
       module_function
 
       def extract_rhs(node)
-        if node.casgn_type?
-          _scope, _lhs, rhs = *node
-        elsif node.op_asgn_type?
-          _lhs, _op, rhs = *node
-        elsif node.call_type?
-          rhs = node.last_argument
-        elsif node.assignment?
-          _lhs, rhs = *node
+        if node.call_type?
+          node.last_argument
+        elsif node.casgn_type? || node.op_asgn_type? || node.assignment?
```

Comment:
`node.assignment?` hanldes `casgn_type?` and `op_asgn_type?`
```suggestion
        elsif node.assignment?
```